### PR TITLE
neo4j: 3.4.10 -> 3.5.2

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
I noticed neo4j in nixpkgs was outdated (most likely because `neo4j-admin` was removed and @r-ryantm couldn't autobump anymore).

Also added a test.

cc @offlinehacker 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/nosql/neo4j/default.nix [hunk 1]
```diff
@@ -4,21 +4,22 @@ with stdenv.lib;
...
 
   buildInputs = [ makeWrapper jre8 which gawk ];
 
+
   installPhase = ''
     mkdir -p "$out/share/neo4j"
     cp -R * "$out/share/neo4j"
 
     mkdir -p "$out/bin"
-    for NEO4J_SCRIPT in neo4j neo4j-admin neo4j-import neo4j-shell cypher-shell
```
Comment (alyssais) [inline pkgs/servers/nosql/neo4j/default.nix]: Guessing this was removed upstream?

File: nixos/tests/neo4j.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import ./make-test.nix {
+  name = "neo4j";
+
+  nodes = {
+    master =
+      { pkgs, ... }:
```
Comment (alyssais) [inline nixos/tests/neo4j.nix]: I'd drop this unused argument.

**Comment by alyssais** [comment]
@GrahamcOfBorg build neo4j

File: pkgs/servers/nosql/neo4j/default.nix (hunk 1 above)
Comment (flokli) [reply pkgs/servers/nosql/neo4j/default.nix]: yup, this was [dropped in 3.4.11](https://neo4j.com/release-notes/neo4j-3-4-11/), and was considered deprecated a while before that.

File: nixos/tests/neo4j.nix (hunk 2 above)
Comment (flokli) [reply nixos/tests/neo4j.nix]: done!

**Pull request merged**